backend/src/bundles/users/user.repository.ts

```diff
@@ -110,6 +113,47 @@ class UserRepository implements Repository {
             .execute();
     }
 
+    public async update(
+        query: Record<string, unknown>,
+        payload: Record<string, Partial<UserDetailsModel>>,
+    ): Promise<UserEntity> {
+        const trx = await this.userModel.startTransaction();
+        const userId = query['id'] as number;
+        const updatedUserDetails = payload[
+            'userDetails'
+        ] as Partial<UserDetailsModel>;
+        try {
+            const user = await this.userModel.query(trx).findById(userId);
+
+            if (!user) {
+                throw new HttpError({
+                    message: UserValidationMessage.USER_NOT_FOUND,
+                    status: HttpCode.NOT_FOUND,
+                });
+            }
```

Comment:
Do we need this check as we already checking users exist stance in auth plugin?

Reply:
Here `const user: UserModel | undefined` so it is need to be checked if it is not undefined.